modules/pubsub/nats/nats.go

```diff
@@ -0,0 +1,41 @@
+package nats
+
+import (
+	nts "github.com/nats-io/nats.go"
+
+	"github.com/spaceuptech/space-cloud/model"
+)
+
+// nats holds the nats driver session
+type nats struct {
+	conn *nts.EncodedConn
+}
+
+func Connect(connection string) (*nats, error) {
+	nc, err := nts.Connect(connection)
+	c, _ := nts.NewEncodedConn(nc, "json")
+	if err != nil {
+		return nil, err
+	}
+	return &nats{c}, nil
+}
+
+func (n *nats) Publish(subject string, msg *model.PubsubMsg) error {
+	sendChannel := make(chan *model.PubsubMsg)
+	// defer sendChannel.Close()
+	err := n.conn.BindSendChan(subject, sendChannel)
+	if err != nil {
+		return err
+	}
+	sendChannel <- msg
+	return nil
+}
+
+func (n *nats) QueueSubscribe(subject, queue string, ch chan *model.PubsubMsg) (func() (error) , error) {
+	natsWildcard := ">"
+	subs, err := n.conn.BindRecvQueueChan(subject + natsWildcard, queue, ch)
+	if err != nil {
+		return nil, err
+	}
+	return subs.Unsubscribe, nil
```

Comment:
Maybe you should return pubsubUnsubscribe from here

Reply:
Okay

Reply:
Not yet resolved

Reply:
Done in the latest commit